blog/articles/set-up-a-local-knative-environment-with-kind.md

```diff
@@ -0,0 +1,315 @@
+
+```bash
+curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.8.1/kind-$(uname)-amd64
+chmod +x ./kind
+mv ./kind /some-dir-in-your-PATH/kind
+```
+
+The next step is to create a Kubernetes cluster using `kind`. While you can certainly use `kubectl port-forward` to get traffic into the cluster, a nicer way is to expose the ports the ingress gateway will listen on to the host. To do that, you can pass in a file with cluster configuration parameters:
+
+```bash
+cat > clusterconfig.yaml <<EOF
+kind: Cluster
+apiVersion: kind.sigs.k8s.io/v1alpha3
+nodes:
+- role: control-plane
+  extraPortMappings:
+	## expose port 31380 of the node to port 80 on the host
+  - containerPort: 31080
+    hostPort: 80
+	## expose port 31443 of the node to port 443 on the host
+  - containerPort: 31443
+    hostPort: 443
+EOF
+```
+
+The values for the container ports are randomly chosen and are used later on to configure a NodePort service with these values. The values for the host ports are where you'll send cURL requests to as you deploy applications to the cluster.
+
+With that cluster configuration file, you can create a cluster with the below command. Your `kubeconfig` will automatically be updated and the default cluster will be set to your new cluster.
+
+```bash
+$ kind create cluster --name knative --config clusterconfig.yaml
+Creating cluster "knative" ...
+ ✓ Ensuring node image (kindest/node:v1.18.2) 🖼
+ ✓ Preparing nodes 📦
+ ✓ Writing configuration 📜
+ ✓ Starting control-plane 🕹️
+ ✓ Installing CNI 🔌
+ ✓ Installing StorageClass 💾
+Set kubectl context to "kind-knative"
+You can now use your cluster with:
+
+kubectl cluster-info --context kind-knative
+
+Have a nice day! 👋
+```
+
+## Knative Serving
+Now that the cluster is running, you can add the Knative components starting with Knative CRDs. At the time of writing the latest available version is `0.15`
+
+```bash
+$ kubectl apply --filename https://github.com/knative/serving/releases/download/v0.15.0/serving-crds.yaml
+
+customresourcedefinition.apiextensions.k8s.io/certificates.networking.internal.knative.dev created
+customresourcedefinition.apiextensions.k8s.io/configurations.serving.knative.dev created
+customresourcedefinition.apiextensions.k8s.io/ingresses.networking.internal.knative.dev created
+customresourcedefinition.apiextensions.k8s.io/metrics.autoscaling.internal.knative.dev created
+customresourcedefinition.apiextensions.k8s.io/podautoscalers.autoscaling.internal.knative.dev created
+customresourcedefinition.apiextensions.k8s.io/revisions.serving.knative.dev created
+customresourcedefinition.apiextensions.k8s.io/routes.serving.knative.dev created
+customresourcedefinition.apiextensions.k8s.io/serverlessservices.networking.internal.knative.dev created
+customresourcedefinition.apiextensions.k8s.io/services.serving.knative.dev created
+customresourcedefinition.apiextensions.k8s.io/images.caching.internal.knative.dev created
+```
```

Comment:
Again would be good to have the command and then the example as different code blocks